### PR TITLE
Fixes #25703 - Fix tests for enforcing taxonomies

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

File: test/controllers/dashboard_controller_test.rb [hunk 1]
```diff
@@ -23,10 +23,12 @@ class DashboardControllerTest < ActionController::TestCase
 
   test '#destroy removes a widget from the user' do
     widget = FactoryBot.create(:widget, :user => users(:admin))
+    assert_includes users(:admin).widget_ids, widget.id
     delete :destroy, params: { :id => widget.id, :format => 'json' }, session: set_session_user
     assert_response :success
     assert_equal widget.id.to_s, @response.body
-    assert_empty users(:admin).widgets.reload
+    users(:admin).widgets.reload
+    refute_includes users(:admin).widget_ids, widget.id
```
Comment (tbrisker) [inline test/controllers/dashboard_controller_test.rb:31]: This one isn't really related but it was failing locally so I fixed it along the way.

**Comment by shlomizadok** [comment]
🔪

**Comment by mmoll** [comment]
test failures related, I guess :broken_heart:

**Comment by tbrisker** [comment]
all :green_apple: now

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
merged, thanks @tbrisker!

**Comment by lzap** [comment]
Holy crap it's real.

Okay, I think this PR accidentally removed tests starting with "#set_interfaces", these do not seem to be taxonomy related. I mean, they were probably in the context where taxonomy was disabled but only for engineer laziness probably. We need them. There's probably more but these are important I think. @ares ?

**Comment by lzap** [comment]
Indeed, I've put all tests back, deleted the taxonomy context and only 5 of those really fail:

```
Failure:
HostTest::location or organizations are not enabled#test_0116_compute attributes are populated by hardware profile from hostgroup [test/models/host_test.rb:3750]:
Organization can't be blank and Location can't be blank

Error:
HostTest::location or organizations are not enabled#test_0023_when setting host environment to nil, its puppet classes should be removed:
NameError: undefined local variable or method `h' for #<#<Class:0x00005555636da430>:0x0000555565fe2940>
    test/models/host_test.rb:2522:in `block (2 levels) in <class:HostTest>'

Failure:
HostTest::location or organizations are not enabled#test_0117_compute attributes are populated by hardware profile passed to host [test/models/host_test.rb:3762]:
Organization can't be blank and Location can't be blank

Failure:
HostTest::location or organizations are not enabled#test_0042_should save uuid on managed hosts [test/models/host_test.rb:2727]:
Expected false to be truthy.

Failure:
HostTest::location or organizations are not enabled#test_0041_should validate pxe loader when provided [test/models/host_test.rb:2720]:
Expected {:organization_id=>["can't be blank"], :location_id=>["can't be blank"]} to be empty.

Finished in 54.126864s, 6.6141 runs/s, 18.5675 assertions/s.
358 runs, 1005 assertions, 4 failures, 1 errors, 0 skips
```

**Comment by tbrisker** [comment]
@lzap oops. I guess that's what happens when we have a file with a gazillion tests, i was assuming we are either testing same behavior both with or without taxonomies or behavior that is specific to disabled taxonomies. Since you already pulled them back, feel like opening a pr with it? the 5 failing tests will need to figure out if they should remain and be fixed or removed.

**Comment by lzap** [comment]
I already have merge conflict on my PR, I suggest to revert the changes which git will understand I hope and then reapply the changes cleanly (without removing those) that should I believe prevent conflicts. Or any other ideas?

**Comment by lzap** [comment]
Another idea - since this test file is way too long, maybe time to break it up into smaller files? Since we are probably caused conflicts already we could do this. Not sure. I can proceed with the PR if there is not any better idea.

**Comment by ares** [comment]
I don't like splitting into smaller files, not sure how I would know how to easily check whether there's some test covering some method already, one test file per model/concern/service class please

**Comment by tbrisker** [comment]
@ares the real problem here is that the host model is too big, so the tests are also too big.
@lzap not sure why you're getting merge conflicts, but if you prefer i'll open a pr to revert just the needed change. I don't want to revert the whole commit since it contains a lot of other useful changes.

**Comment by lzap** [comment]
Let me do a PR then, I am already on it.

**Comment by lzap** [comment]
Still working, still red. Tomorrow.

**Comment by lzap** [comment]
Ok this is probably my last work item of the year, have fun:

https://github.com/theforeman/foreman/pull/6362

Ran out of time, please take from there. I only needed like 10 minutes.

**Pull request merged**